packages/@aws-cdk/core/test/test.cfn-resource.ts

```diff
@@ -124,6 +124,35 @@ export = nodeunit.testCase({
       // No DependsOn!
     });
 
+    test.done();
+  },
+
+  'CfnResource cannot be created outside Stack'(test: nodeunit.Test) {
+    const app = new core.App();
+    test.throws(() => {
+      new core.CfnResource(app, 'Resource', {
+        type: 'Some::Resource',
+      });
+    }, /No stack could be identified for the construct/);
```

Comment:
This is likely 'new to the CDK' error and the message for someone like that is a bit vague.

Either something a bit more clear to tell the user that they've done something incorrectly or an action to correct it may help.